Last Week in Pony - June 18, 2023 (#865)
ponylang/ponylang-website: content/blog/last-week-in-pony-061823.md

<< content >>

If you'd be interested in attending an Office Hours in the future, you should join some time, there's a [calendar you can subscribe to](https://calendar.google.com/calendar/ical/4465e68ae24131ae00461a40893f2637a2c9ac510e311a44ff78680e2f183ce3%40group.calendar.google.com/public/basic.ics) to stay up-to-date with the schedule. We do our best to keep the calendar up-to-date.

## Community Resource Highlight

We like to take a moment in each Last Week in Pony to highlight a community resource. There are many community resources that can go unappreciated until _just the right time_ when someone hops into the Ponylang Zulip asking a question or facing a problem we have all had at one time or another. Well here in Last Week in Pony, we make it **just the right time** to highlight one of our excellent community resources.

This week we are highlighting the [`ponylang/fork_join`](https://github.com/ponylang/fork_join) library.

> There was a time when many folks would sign up for the [Pony Zulip](https://ponylang.zulipchat.com/) and ask how they could do something like us Pony to concurrently process a large array or something similar and collect the results. This class of problems falls under the [fork-join model](https://en.wikipedia.org/wiki/Fork%E2%80%93join_model).

Reviewer: ```suggestion
There was a time when many folks would sign up for the [Pony Zulip](https://ponylang.zulipchat.com/) and ask how they could use Pony to concurrently process a large array or something similar and collect the results. This class of problems falls under the [fork-join model](https://en.wikipedia.org/wiki/Fork%E2%80%93join_model).
```

> The `ponylang/fork_join` library hands the "boilerplate" needed for doing the basic book keeping needed to split up a job into multiple parts and collect the results. By default, work done using the `fork_join` will be split up across X `Worker` actors where X is equal to the number of scheduler threads available to the program.

Reviewer: ```suggestion
The `ponylang/fork_join` library handles the "boilerplate" needed for doing the basic bookkeeping needed to split up a job into multiple parts and collect the results. By default, work done using the `fork_join` will be split up across X `Worker` actors where X is equal to the number of scheduler threads available to the program.
```
